### PR TITLE
Flag "echo" transfers as imitations

### LINKED ISSUE #2025: Flag "echo" incoming transfers
### Description

We flag spoof events as imitation transactions. We don't, however, flag the other address poisoning vector whereby an attacker "echoes" an outgoing transfer with an incoming one of the same token.

We should flag the above transactions if they are under a certain limit.

### Requirements

Flag incoming transfers that following outgoing transfers of the same token address, under a defined limit.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by iamacook**
## Summary

Resolves #2025

An address poisoning vector we don't flag are incoming transfers of the same asset sent of minimal value, or "echoes". When sending a large amount of an asset, it is common to receive a low value of the same asset from a malicious actor in the hopes you copy their address and send to them instead.

This adds new logic to flag such transactions. If the receipt of a token occurs after it being sent, under a defined limit, it is deemed an "echo" and flagges as such.

## Changes

- Add `mappings.imitation.echoLimit` value
- Flag "echo" transactions
- Add appropriate test coverage

### CODE REVIEW

File: src/routes/transactions/mappers/transfers/transfer-imitation.mapper.ts [hunk 1]
```diff
@@ -113,7 +117,10 @@ export class TransferImitationMapper {
           return false;
         }
 
-        return this.isSpoofedEvent(txInfo, prevTxInfo);
+        return (
+          this.isSpoofedEvent(txInfo, prevTxInfo) ||
+          this.isEchoImitation(txInfo, prevTxInfo)
+        );
```
Comment (hectorgomezv) [inline src/routes/transactions/mappers/transfers/transfer-imitation.mapper.ts:123]: (This is more a general comment, no directly related to the PR code, and I think it's out of the scope) 

As this `TransferImitationMapper` gets more complex, it is likely getting more computationally expensive. I think it's fine for now, but given this mapper is applied to each group of transactions, and if I'm not wrong it's deterministic, it'd be great to somehow cache/memoize whether a given transaction is an imitation one. Wdyt?

**Review by hectorgomezv — APPROVED** [review]
👏🏻👏🏻👏🏻

File: src/routes/transactions/mappers/transfers/transfer-imitation.mapper.ts (hunk 1 above)
Comment (iamacook) [reply src/routes/transactions/mappers/transfers/transfer-imitation.mapper.ts:123]: I agree - we should look into caching the mapped entities (potentially across the project). I've noted it down for discussion.

File: src/routes/transactions/transactions-history.imitation-transactions.controller.spec.ts [hunk 2]
```diff
@@ -105,6 +107,9 @@ describe('Transactions History Controller (Unit) - Imitation Transactions', () =
     await app.close();
   });
 
+  const chain = chainBuilder().build();
+  const safe = safeBuilder().build();
```
[the remark this answers is not in the document]
Comment (iamacook) [reply src/routes/transactions/transactions-history.imitation-transactions.controller.spec.ts]: Moved in f17faf5668b87975a98654441009908a61a0d205.

**Review by hectorgomezv — APPROVED** [review]

**Pull request merged**